src/Controller/AppController.php

```diff
@@ -100,6 +100,8 @@ public function beforeRender(Event $event) : void
 
             $this->set(compact('user'));
         }
+
+        $this->viewBuilder()->setTemplatePath('Pages/' . $this->name);
```

Comment:
This can probably be achieved by overriding Controller's `_viewPath` method, thereby making an explicit call to `setTemplatePath` not needed:

```php
class AppController extends Controller
{
    // ...
    protected function _viewPath()
    {
        return sprintf('Pages/%s', parent::_viewPath());
    }
}
```